### DISCUSSION TITLE
pip install -e . results in AttributeError: module 'pkgutil' has no attribute 'ImpImporter with py3.12

### DISCUSSION THREAD

**Discussion by saiaprameya** [pypa/setuptools#4107] [Q&A] [answered]
While trying to run pip install -e . results in the below said error    

Python 3.12.0

pip list | grep setuptools
setuptools           65.7.0

 Traceback (most recent call last):
      File "<string>", line 2, in <module>
      File "<pip-setuptools-caller>", line 14, in <module>
      File "virt-py12/lib/python3.12/site-packages/setuptools/__init__.py", line 16, in <module>
        import setuptools.version
      File "virt-py12/lib/python3.12/site-packages/setuptools/version.py", line 1, in <module>
        import pkg_resources
      File "virt-py12/lib/python3.12/site-packages/pkg_resources/__init__.py", line 2191, in <module>
        register_finder(pkgutil.ImpImporter, find_on_path)
                        ^^^^^^^^^^^^^^^^^^^
    AttributeError: module 'pkgutil' has no attribute 'ImpImporter'. Did you mean: 'zipimporter'?

**Comment by abravalheri** [comment]
This problem is likely to have been fixed in the latest versions of setuptools. 

Please note that if the library your are trying to install limite the version of setuptools to an older value, you might have to contact that library's devs.

**Reply by saiaprameya** [reply]
Thanks Anderson for the confirmation. Can you please let me know which version has this change. Will try to install the same.

**Reply by abravalheri** [reply, marked as answer]
I recommend using the latest version.

Setuptools release process is linear (monotonic increasing versions) and we don't have bandwidth for maintaining multiple versions at the same time. So all bugs are corrected in the `main` branch and released in the subsequent version. Unfortunately backporting the bug fixes to old versions is not an option for us.
